apps/server/src/modules/files-storage/entity/filerecord.entity.ts

```diff
@@ -158,26 +158,70 @@ export class FileRecord extends BaseEntity {
 		this.deletedSince = props.deletedSince;
 	}
 
-	updateSecurityCheckStatus(status: ScanStatus, reason: string): void {
+	public updateSecurityCheckStatus(status: ScanStatus, reason: string): void {
 		this.securityCheck.status = status;
 		this.securityCheck.reason = reason;
 		this.securityCheck.updatedAt = new Date();
 		this.securityCheck.requestToken = undefined;
 	}
 
-	markForDelete(): void {
+	public markForDelete(): void {
 		this.deletedSince = new Date();
 	}
 
-	unmarkForDelete(): void {
+	public unmarkForDelete(): void {
 		this.deletedSince = undefined;
 	}
 
-	setName(name: string): void {
+	public setName(name: string): void {
 		if (name.length === 0) {
 			throw new BadRequestException(ErrorType.FILE_NAME_EMPTY);
 		}
 
 		this.name = name;
 	}
+
+	public hasSameName(name: string): boolean {
+		const hasSameName = this.name === name;
+
+		return hasSameName;
+	}
+
+	public getName(): string {
+		return this.name;
+	}
+
+	public isBlocked(): boolean {
+		const isBlocked = this.securityCheck.status === ScanStatus.BLOCKED;
+
+		return isBlocked;
+	}
+
+	public isPending(): boolean {
+		const isPending = this.securityCheck.status === ScanStatus.PENDING;
+
+		return isPending;
+	}
+
+	public isVerified(): boolean {
+		const isVerified = this.securityCheck.status === ScanStatus.VERIFIED;
+
+		return isVerified;
+	}
+
+	public getParentDescriptions(): { parentType: FileRecordParentType; parentId: EntityId } {
+		const { parentId, parentType } = this;
+
+		return { parentId, parentType };
+	}
+
+	public getDescriptions(): { size: number; name: string; mimeType: string } {
+		const { size, name, mimeType } = this;
+
+		return { size, name, mimeType };
+	}
+
+	public getSchoolId(): EntityId {
```

Comment:
Kann in der paths.ts und files-storage.mapper.ts auch noch verwendet werden.

Reply:
Bei paths.ts eingeführt. Bei mapper mit Absicht erstmal nicht. Es kann sein das für mapper aller Art ein greife immer auf props zu gilt, oder ein getProperties(); getter kommt. Schaue ich mir im do mapper ticket an.